lightning/src/routing/router.rs

```diff
@@ -962,6 +962,43 @@ impl PaymentParameters {
 	}
 }
 
+/// Information manually provided by the user to route the payment
+#[derive(Clone, Copy)]
+pub struct ManualRoutingParameters {
+	/// Same as [`RouteParameters::max_total_routing_fee_msat`]
+	pub max_total_routing_fee_msat: Option<u64>,
+	/// Same as [`PaymentParameters::max_total_cltv_expiry_delta`]
+	pub max_total_cltv_expiry_delta: Option<u32>,
+	/// Same as [`PaymentParameters::max_path_count`]
+	pub max_path_count: Option<u8>,
+	/// Same as [`PaymentParameters::max_channel_saturation_power_of_half`]
+	pub max_channel_saturation_power_of_half: Option<u8>,
+}
+
+impl_writeable_tlv_based!(ManualRoutingParameters, {
+	(1, max_total_routing_fee_msat, option),
+	(3, max_total_cltv_expiry_delta, option),
+	(5, max_path_count, option),
+	(7, max_channel_saturation_power_of_half, option),
+});
+
+impl ManualRoutingParameters {
+	/// Initates an empty set of [`ManualRoutingParameters`]
+	pub fn new() -> Self {
```

Comment:
Random though: We would like to use an `fn empty` name for this and keep the `new` for a common constructor (that I do not know what common in this case means 😄 )

Reply:
That makes sense!

I noticed, in the codebase, we're using `fn new` more often for blank initialization, though, so to keep things consistent, it might be a good idea to stick with `new`. Lemmino what you think! :)